StreetHawk/Classes/Core/Publish/SHUtils.m

```diff
@@ -121,6 +121,11 @@ void SHLog(NSString *format, ...)
         }
         if (out == nil)
         {
+            [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSSZ"];
```

Comment:
a better way to do this is to have a list of formats and loop through them and return when out!=nill.

That way if you want to add a new format you just add to the list.

that will make you code 90% smaller than it is now.

Reply:
I think @StreethawkNick already mentioned this to you last time you added another conditional for a new format.